strapi/strapi
# [DEITS] Fix telemetry by waiting for the events to end before exiting process

### What does it do?

Fix the telemetry didDEITSProcessFinish event by waiting for the telemetry finish event to send before exiting the CLI process.

### Why is it needed?

After `engine.transfer()` there was nothing to give the events time to send before the process was exiting.

Note: I've simplified it greatly from the previous fix. Sad not to be able to make use of the event listeners, but I think this is preferable to any hacky workarounds.

### How to test it?

Run import and export and the telemetry "didDEITSProcessFinish" events should always send.
